index.php

```diff
@@ -58,10 +53,45 @@
         </form>
     </header>
 
-
-    <form method="POST">
+    <div>
+        <h3>Filter by car brand</h3>
+        <form method="POST">
+
+            <?php
+
+            $model = new CarModel(make_db());
+            if (isset($_POST["car_brand_filter"])) {
+                $brand = $_POST["car_brand_filter"];
+
+                if (empty($brand)) {
+                    $cars = $model->getAllCarInfo();
+                } elseif (
+                    $model->validateCarBrand($brand) ||
+                    is_string($brand) ||
+                    strlen($brand) <= 20 ||
+                    strlen($brand) > 0
+                ) {
+                    $cars = $model->filterCarBrand($brand);
+                } else {
+                    echo "Invalid brand name";
+                }
+                if (!empty($cars)) {
+                    $car_list = create_list_of_cars($cars);
+                }
+            } else {
+                $cars = $model->getAllCarInfo();
+                $car_list = create_list_of_cars($cars);
+            }
```

Comment:
This is quite complicated. What if you made it so there just a single method to get cars that can be optionally filtered - that way you don't have to decide which one to call. 

It would simplify this code drastically